src/preload/preload.js

```diff
@@ -40,11 +40,39 @@ htmx.defineExtension('preload', {
         }
 
         // Special handling for HX-GET - use built-in htmx.ajax function
-        // so that headers match other htmx requests, then set
+        // so that headers match other htmx requests, then intercept the request
+        // to extract the request parameters and perform XMLHttpRequest directly
+        // to avoid creating any events and making side effects. Afterward, set
         // node.preloadState = TRUE so that requests are not duplicated
         // in the future
         var hxGet = node.getAttribute('hx-get') || node.getAttribute('data-hx-get')
         if (hxGet) {
+          // function that intercepts htmx.ajax requests and performs them
+          // with XMLHttpRequest directly to avoid any side effects
+          node.prefetchEventHandler = function(event) {
+            node.removeEventListener('htmx:beforeRequest', node.prefetchEventHandler, true)
+            event.stopImmediatePropagation()
+            event.preventDefault()
+            // Since it is a GET request, we're only interested in the 
+            // request headers and the URL
+            const headers = event.detail.requestConfig.headers
+            const url = event.detail.pathInfo.finalRequestPath
+
+            const r = new XMLHttpRequest()
```

Comment:
Didn't test this at all, but I'm wondering if it could work and be way easier here to use htmx's own `XMLHTTPRequest` that is passed in the `htmx:beforeRequest` event details ? This way you would probably not even need to copy the headers at all

See https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L4342

Reply:
It looks like this xhr object has a few callbacks attached to it, and they will trigger unwanted htmx events if I would reuse htmx's own xhr.

Reply:
Looked into the htmx code, indeed the xhr is assigned 4 callback before firing `htmx:beforeRequest`
```
xhr.onload = function() {...}
xhr.onerror = function() {...}
xhr.onabort = function() {...}
xhr.ontimeout = function() {...}
```

You already override `onload` here, I'm thinking it could be as easy as assigning the 3 others to null or an empty function, to reuse the XHR without any side effect ?
Other listeners are bound but [after the `htmx:beforeRequest` event](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L4418) so you shouldn't need to handle those.

I don't have any strong opinion on the matter tbh, just wanted to emphasize this possibility!
If you prefer creating a new XHR then I'm perfectly fine with it too